### PR TITLE
feat: Add aria APIs to Field base class

### PR DESCRIPTION
**Pull request by mikeharv**
## The basics

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

### Resolves

Fixes https://github.com/RaspberryPiFoundation/blockly/issues/9654

### Proposed Changes

This PR introduces new ARIA helper APIs on the base `Field` class to support accessible label generation:

- Adds:
  - `getAriaTypeName(): string | null`
  - `getAriaValue(): string | null`
  - `computeAriaLabel(includeTypeInfo?: boolean): string`
- Extends `FieldConfig` with optional `ariaTypeName` to allow block definitions to configure ARIA type labels
- Implements safe default behavior in the base class:
  - `getAriaTypeName` returns configured value (if present)
  - `getAriaValue` falls back to `getValue()` stringification
  - `computeAriaLabel` composes type/value with optional verbosity
- Adds unit tests covering:
  - Default behavior
  - Config-driven type names
  - Label composition logic
  - Null/undefined edge cases
  - Subclass override behavior

These changes establish a consistent foundation for ARIA labeling across all fields.

### Reason for Changes

Blockly currently lacks a standardized way for fields to expose ARIA-friendly labels. 

### Test Coverage

Added a new Aria test suite covering:

- `getAriaTypeName`
  - Default (`null`)
  - JS constructor config
  - JSON config
- `getAriaValue`
  - Strings, numbers
  - `null` and `undefined`
- `computeAriaLabel`
  - Value-only output
  - Type + value composition
  - Type-only cases
  - Empty output when both are missing
- Subclass overrides
  - Custom `getAriaValue`
  - Custom `getAriaTypeName`
  - Combined overrides

Tests ensure correct fallback behavior and verify that subclasses can override these methods as intended.

### Documentation

Inline JSDoc comments were added for all new APIs.

No external documentation updates yet.

### Additional Information

Follow-up PRs are planned to adopt the new APIs across built-in fields.

### CODE REVIEW

File: packages/blockly/core/field.ts [hunk 1]
```diff
@@ -300,6 +306,88 @@ export abstract class Field<T = any>
...
+
+  /**
+   * Gets an ARIA-friendly label representation of this field's value.
+   *
+   * Note that implementations should generally always override this value to
+   * ensure a non-null value is returned since the default implementation relies
+   * on 'getValue' which may return null, and a null return value for this
+   * function will prompt ARIA label generation to skip the field's value
+   * entirely when there may be a better contextual placeholder to use, instead,
+   * specific to the field.
```
Comment (maribethb) [inline packages/blockly/core/field.ts:330]: I think it might be useful here to include some examples of potential overrides, to more clearly illustrate the point. Something like:

> For example, a text input field may have a value of `null` if no text has been entered. To avoid hiding this field from screenreaders, implementations should ensure that if the value is `null`, this function would return an appropriate, localized, value such as "Empty text"

We'll have written docs eventually explaining what custom field authors need to do but having it here would probably help too.

**Review by maribethb — APPROVED** [review]
